src/format/opamFile.ml

```diff
@@ -1361,6 +1365,9 @@ module ConfigSyntax = struct
       if t.switch = None then with_switch sw t
       else Pp.bad_format "Multiple switch specifications"
     in
+    let with_previous_switch sw t =
+      with_previous_switch (Some sw) t
+    in
```

Comment:
There is no specific behavior added, better raise it up, with `with_previous_switch`.